x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java

```diff
@@ -0,0 +1,360 @@
+        String field = randomStringField(previousOutput);
+        if (field == null || policies.isEmpty()) {
+            return "";
+        }
+        // TODO add WITH
+        return " | enrich " + randomFrom(policies).policyName() + " on " + field;
+    }
+
+    private static String grok(List<Column> previousOutput) {
+        String field = randomStringField(previousOutput);
+        if (field == null) {
+            return "";// no strings to grok, just skip
+        }
+        StringBuilder result = new StringBuilder(" | grok ");
+        result.append(field);
+        result.append(" \"");
+        for (int i = 0; i < randomIntBetween(1, 3); i++) {
+            if (i > 0) {
+                result.append(" ");
+            }
+            result.append("%{WORD:");
+            if (randomBoolean()) {
+                result.append(randomAlphaOfLength(5));
+            } else {
+                result.append(randomName(previousOutput));
+            }
+            result.append("}");
+        }
+        result.append("\"");
+        return result.toString();
+    }
+
+    private static String dissect(List<Column> previousOutput) {
+        String field = randomStringField(previousOutput);
+        if (field == null) {
+            return "";// no strings to dissect, just skip
+        }
+        StringBuilder result = new StringBuilder(" | dissect ");
+        result.append(field);
+        result.append(" \"");
+        for (int i = 0; i < randomIntBetween(1, 3); i++) {
+            if (i > 0) {
+                result.append(" ");
+            }
+            result.append("%{");
+            if (randomBoolean()) {
+                result.append(randomAlphaOfLength(5));
+            } else {
+                result.append(randomName(previousOutput));
+            }
+            result.append("}");
+        }
+        result.append("\"");
+        return result.toString();
+    }
+
+    private static String keep(List<Column> previousOutput) {
+        int n = randomIntBetween(1, previousOutput.size());
+        Set<String> proj = new HashSet<>();
+        for (int i = 0; i < n; i++) {
+            proj.add(randomName(previousOutput));
+        }
+        return " | keep " + proj.stream().collect(Collectors.joining(", "));
+    }
+
+    private static String randomName(List<Column> previousOutput) {
+        return previousOutput.get(randomIntBetween(0, previousOutput.size() - 1)).name();
+    }
+
+    private static String rename(List<Column> previousOutput) {
+        int n = randomIntBetween(1, Math.min(3, previousOutput.size()));
+        Set<String> proj = new HashSet<>();
+        List<String> names = previousOutput.stream().map(Column::name).collect(Collectors.toList());
+        for (int i = 0; i < n; i++) {
+            String name = names.remove(randomIntBetween(0, names.size() - 1));
+            proj.add(name + " AS " + randomAlphaOfLength(5));
```

Comment:
Another idea is to rename as an existing name. I don't remember what restrictions we have in place for this but some food for thought maybe.